rust-runtime/aws-smithy-http-server/src/macros.rs

```diff
@@ -171,5 +171,50 @@ macro_rules! composite_rejection {
                 }
             }
         }
+    }
+}
+
+macro_rules! opaque_future {
```

Comment:
If you add a small doc about what is an opaque future, it will reflect into cargo docs, which is nice.